app/views/sessions/new.html.erb

```diff
@@ -4,10 +4,10 @@
     <h1>Log In</h1>
     <%= form_for @user, url: "/sessions", method: "post" do |f| %>
         <div class="form-group">
-          <%= f.email_field :email, placeholder: "Email", class: "form-control" %>
+          <%= f.email_field :email, placeholder: "Email", required: true, class: "form-control" %>
         </div>
         <div class="form-group">
-          <%= f.password_field :password, placeholder: "Password", class: "form-control" %>
+          <%= f.password_field :password, placeholder: "Password", required: true, class: "form-control" %>
```

Comment:
You can also validate length of password – please do for all password fields in your templates.

Reply:
Cool! I forgot that. will do now!

Reply:
I just notice that I did the validate length of password at user model.